c10/core/ScalarType.h

```diff
@@ -592,55 +588,78 @@ static inline ScalarType promoteTypes(ScalarType a, ScalarType b) {
     return ScalarType::Undefined;
   }
 
-  // Bits and Quantized are 7 dtypes already handled and not included
-  // in the promotion table below. Therefore every dtype above them
-  // needs to be shifted down to account for that.
-  static constexpr int shift_distance =
-      static_cast<int>(ScalarType::Float8_e5m2) -
-      static_cast<int>(ScalarType::QUInt4x2);
-  static constexpr int shift_threshold = static_cast<int>(ScalarType::Bits16);
-
-  if (static_cast<int>(a) > shift_threshold) {
-    a = static_cast<ScalarType>(static_cast<int>(a) - shift_distance);
+  if (isFloat8Type(a) && a == b) {
```

Comment:
Why don't we have a condition at the top that is
```c++
if (a == b) {
  return a;
}
```

Reply:
Cause we want that diagonal to be doing something... lol I can add this

Reply:
I guess but it makes all of these conditions so weird :p